theoplayer/how-to-guides/01-ads/13-custom-ssai-integration.md

```diff
@@ -0,0 +1,127 @@
+# Custom Server Side Ad Insertion
+
+THEOplayer provides a way to integrate with third-party advertisement providers,
+and have them report their ads and ad-related events through the THEOplayer APIs.
+
+## Android SDK
+
+### Prerequisites
+
+Your THEOplayer SDK needs to be on version 7.6.0 or higher.
+
+### Integration ID
+
+To make use of the custom server side ad insertion, first you need an integration ID which will identify the provided sources and ads to the specific ad provider.
+
+### CustomSsaiDescriptionSerializer
+
+To play a source, inheriting the `CustomSsaiDescription` class is needed that will contain the necessary properties for the specific integration and be passed in the SourceDescription.
+Then, register a `CustomSsaiDescriptionSerializer` using the integration ID and provide a way to serialize/deserialize the new
+`CustomSsaiDescription`.
+
+Below is an example of a YoSpace integration showing `YospaceSsaiDescription` extending `CustomSsaiDescription`:
+
+```kotlin
+data class YospaceSsaiDescription constructor(
+    /**
+     * The type of the requested stream.
+     *
+     * Default: [YospaceStreamType.LIVE]
+     */
+    val streamType: YospaceStreamType = YospaceStreamType.LIVE,
+    /**
+     * Custom properties to set when initializing the Yospace session.
+     */
+    val sessionProperties: Session.SessionProperties = Session.SessionProperties()
+) : CustomSsaiDescription() {
+    override val customIntegration = YospaceConnector.INTEGRATION_ID
+}
+```
+
+And then, the implementation of the `CustomSsaiDescriptionSerializer` specific for YoSpace:
+
+```kotlin
+class YospaceSsaiDescriptionSerializer : CustomSsaiDescriptionSerializer {
+    override fun fromJson(json: String): YospaceSsaiDescription {
+        return Json.decodeFromString(json)
+    }
+
+    override fun toJson(value: CustomSsaiDescription): String {
+        return Json.encodeToString(value)
+    }
+}
+```
+
+Finally, registering the `YospaceSsaiDescriptionSerializer`:
```

Comment:
We should probably explain that this should ideally be registered during initialization of the `CustomSsaiDescription` class (e.g. inside an `init` block of its `companion object`), or at the very least right boefre registering the `ServerSideAdIntegrationFactory`.